File: src/main/java/um/haberes/report/model/haberes/core/CursoCargoDto.java
package um.haberes.report.model.haberes.core;

import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class CursoCargoDto {

    private Long cursoCargoId;

    private Long cursoId;

    private int anho = 0;

    private int mes = 0;

    private Integer cargoTipoId;

    private Long legajoId;

    private BigDecimal horasSemanales = BigDecimal.ZERO;

    private BigDecimal horasTotales = BigDecimal.ZERO;

    private Integer designacionTipoId;

    private Integer categoriaId;

    private byte desarraigo = 0;

    private Long cursoCargoNovedadId;

    private CursoDto curso;

    private CargoTipoDto cargoTipo;

    private PersonaDto persona;

    private DesignacionTipoDto designacionTipo;

    private CategoriaDto categoria;

}

File: src/main/java/um/haberes/report/kotlin/dto/haberes/core/CursoCargoDto.kt
package um.haberes.report.kotlin.dto.haberes.core

import java.math.BigDecimal

data class CursoCargoDto(

    var cursoCargoId: Long? = null,
    var cursoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var cargoTipoId: Int? = null,
    var legajoId: Long? = null,
    var horasSemanales: BigDecimal = BigDecimal.ZERO,
    var horasTotales: BigDecimal = BigDecimal.ZERO,
    var designacionTipoId: Int? = null,
    var categoriaId: Int? = null,
    var desarraigo: Byte = 0,
    var cursoCargoNovedadId: Long? = null,
    var curso: CursoDto? = null,
    var cargoTipo: CargoTipoDto? = null,
    var persona: PersonaDto? = null,
    var designacionTipo: DesignacionTipoDto? = null,
    var categoria: CategoriaDto? = null

)
